test/targets.cpp

```diff
@@ -50,4 +51,21 @@ TEST_CASE(targets)
     EXPECT(ts.size() >= 1);
 }
 
+TEST_CASE(concurrent_targets)
+{
+    std::vector<std::thread> threads;
+
+    for(auto i = 0u; i < 1000; i++)
+    {
+        auto thread_body = []() {
+            auto ref_target = migraphx::make_target("ref");
```

Comment:
target needs to be registered first before it can be made. I think it works right now because it was already registered by other tests.

Reply:
> target needs to be registered first before it can be made (or alternatively targets can by loaded dynamically). I think it works right now because it was already registered by other tests.

It is registered by auto_register.hpp during program initialization, before the test starts.  I wouldn't have another way to make a target instance without calling `make_target()` first.

Recall that the goal of this test is not to make targets correctly; it's to check thread safety.

Reply:
> and then call `unregister_target` at the end.

That's not allowed.  `unregister_target` is called automatically on exit, and it throws an exception if it's called twice with the same target name.  As it is, it's still possible in principle for one holder of a target to unregister it while others are still holding it.

Is there any use at all in having `unregister_target`?  It can only be used at program cleanup.